lib/openhab/log/logger.rb

```diff
@@ -148,38 +152,26 @@ class << self
       #
       # @return [Logger] for the supplied name
       #
-      def logger(klass)
+      def logger_for_class(klass)
         if klass.respond_to?(:java_class) &&
            klass.java_class &&
            !klass.java_class.name.start_with?('org.jruby.Ruby')
           klass = klass.java_class
         end
-        name = klass.name
+        name = klass.name if klass.name != 'Object'
         @loggers[name] ||= Log.logger_for(name)
```

Comment:
This memorization doesn't work because the class name can be shared between rules and therefore the logging won't be correct if they try and log for a specific rule.  The logger name needs to be calculated on each call.

Reply:
I moved the memorisation into `logger_for` just right before the actual Logging object is created. It should cover all cases, since the key is the actual `log_prefix` itself.